app/main/posts/views/filters/filter-saved-search.directive.js

```diff
@@ -17,28 +17,34 @@ function FilterSavedSearch(SavedSearchEndpoint, _,  $rootScope, ModalService) {
         scope.searchesLength = 0;
         scope.loading = false;
         scope.$on('savedSearch:update', loadSavedSearches);
-
+        scope.$watch(PostFilters.getModeId, function (newValue, oldValue) {
```

Comment:
We need to be aware of the modeid changes because that is used across the app to select saved searches and reset them .